extensions/cornerstone/src/tools/CalibrationLineTool.ts

```diff
@@ -81,20 +85,48 @@ export function onCompletedCalibrationLine(servicesManager, csToolsEvent) {
   const currentColumnPixelSpacing =
     calibratedPixelSpacing?.[1] || imagePlaneModule?.columnPixelSpacing || 1;
 
+  /**
+   * calibration action handler
+   *
+   **/
   const adjustCalibration = newLength => {
     const spacingScale = newLength / length;
     const rowSpacing = spacingScale * currentRowPixelSpacing;
     const colSpacing = spacingScale * currentColumnPixelSpacing;
 
+    const renderingEngine = viewport.getRenderingEngine();
+
     // trigger resize of the viewport to adjust the world/pixel mapping
-    calibrateImageSpacing(
-      imageId,
-      viewport.getRenderingEngine(),
-      rowSpacing,
-      colSpacing
-    );
+    calibrateImageSpacing(imageId, renderingEngine, rowSpacing, colSpacing);
+
+    // NOTE: in the StackViewport, if the row and colum pixel spacing from
+    // 'calibratedPixelSpacing' and the 'imagePlaneModule' metadata are different,
+    // it triggers the "IMAGE_SPACING_CALIBRATED" event, which applies calibration
+    // multiple times to the annotations.
+    // So, let's override imagePlaneModule.
+    // This override should happen in the next event cycle, as the rendering is
+    // going on after above function call.
+    setTimeout(() => {
```

Comment:
@md-prog  - please update this change to be the one from FUBerlin - they are contributing that code back to the OSS, but we need to use the final version that is compatible with the released CS3D